pyccel/ast/functionalexpr.py

```diff
@@ -16,10 +16,27 @@
 )
 
 #==============================================================================
-class FunctionalFor(Basic):
+class FunctionalFor(PyccelAstNode):
 
-    """."""
-    __slots__ = ('_loops','_expr', '_lhs', '_indices', '_index')
+    """
+    Represents any generator expression e.g:
+    a = [i for i in range(10)]
+
+    Parameters
+    ----------
+    loops   : CodeBlock/For
+              The loops contained in the expression
+    expr    : Basic
+              The expression at the origin of the expression
+              E.g. 'i' for '[i for i in range(10)]'
+    lhs     : Variable
+              The variable to which the result is assigned
+    indices : list of Variable
+              All iterator targets for the for loops
+    index : ?
```

Comment:
Is this parameter used at all?

Reply:
Not that I could find

Reply:
I suggest that we simply remove it, then

Reply:
Done cf6b45a (hopefully this won't break anything)

Reply:
It turns out this is necessary for expressions such as:
```
a = [i for i in range(2,10,3)]
```
I have put it back and added the description to the docstring : 5adc439b